src/auth.rs

```diff
@@ -0,0 +1,94 @@
+use base64::DecodeError;
+use std::string::FromUtf8Error;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum InvalidBase64 {
+    InvalidByte { index: usize, offending_byte: u8 },
+    InvalidLength,
+    InvalidLastSymbol { index: usize, offending_byte: u8 },
+}
+
+impl From<base64::DecodeError> for InvalidBase64 {
+    fn from(err: base64::DecodeError) -> Self {
+        match err {
+            DecodeError::InvalidByte(index, offending_byte) => InvalidBase64::InvalidByte {
+                index,
+                offending_byte,
+            },
+            DecodeError::InvalidLength => InvalidBase64::InvalidLength,
+            DecodeError::InvalidLastSymbol(index, offending_byte) => {
+                InvalidBase64::InvalidLastSymbol {
+                    index,
+                    offending_byte,
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, thiserror::Error)]
+pub enum AuthorizationError {
+    #[error("The provided base64 is invalid")]
+    InvalidBase64(InvalidBase64),
+    #[error("The decoded UTF-8 is invalid")]
+    InvalidUTF8(FromUtf8Error),
+    #[error("The authentication format was not Basic")]
```

Comment:
Erorr messages in Rust are lowercase, also adds the original error to the error message
```suggestion
    #[error("the provided base64 is invalid: {0}")]
    InvalidBase64(InvalidBase64),
    #[error("the decoded UTF-8 is invalid: {0}")]
    InvalidUTF8(FromUtf8Error),
    #[error("the authentication format was not Basic")]
```